test/jdk/javax/swing/JTable/JTableRightOrientationTest.java

```diff
@@ -0,0 +1,210 @@
+ * questions.
+ */
+
+import java.awt.ComponentOrientation;
+import java.awt.Dimension;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.AbstractTableModel;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import javax.imageio.ImageIO;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * @test
+ * @key headful
+ * @bug 5108458
+ * @summary Test to check Right alignment of JTable data
+ * (Fix affects all L&F, test verifies only Metal L&F)
+ * @run main JTableRightOrientationTest
+ */
+
+public class JTableRightOrientationTest {
+    static JFrame frame;
+    static CustomTable customTableObj;
+    static volatile int allColumnWidths;
+    static volatile Point tableLocation;
+    static volatile Dimension tableSize;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            System.out.println("Testing LAF : " + laf.getClassName());
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    frame = new JFrame("Test JTable");
+                    JPanel panel = new JPanel(new GridBagLayout());
+                    frame.setContentPane(panel);
+                    customTableObj = new CustomTable();
+                    panel.add(new JScrollPane(customTableObj.table),
+                            new GridBagConstraints(0, 0, -1, -1, 1.0, 1.0,
+                                    GridBagConstraints.PAGE_START, GridBagConstraints.BOTH,
+                                    new Insets(2, 2, 2, 2), 0, 0));
+                    frame.pack();
+                    frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+                    frame.applyComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+                    frame.setVisible(true);
+                });
+
+                robot.waitForIdle();
+                robot.delay(1000);
+                SwingUtilities.invokeAndWait(() -> {
+                    allColumnWidths = 0;
+                    for (int i = 0; i < customTableObj.table.getColumnCount(); i++) {
+                        allColumnWidths += customTableObj.table.getTableHeader().getColumnModel()
+                                .getColumn(i)
+                                .getWidth();
+                    }
+                    tableLocation = customTableObj.table.getLocationOnScreen();
+                    tableSize = customTableObj.table.getSize();
```

Comment:
Perform all the calculation here and publish the object with the result:

```suggestion
                    int allColumnWidths = 0;
                    for (int i = 0; i < customTableObj.table.getColumnCount(); i++) {
                        allColumnWidths += customTableObj.table.getTableHeader().getColumnModel()
                                .getColumn(i)
                                .getWidth();
                    }
                    Point tableLocation = customTableObj.table.getLocationOnScreen();
                    Dimension tableSize = customTableObj.table.getSize();
                    tableSize.width -= allColumnWidths;

                    tableBounds = new Rectangle(tableLocation, tableSize);
```
where `tableBounds` is a `static volatile` field of type `Rectangle` that needs to be declared instead of the three fields `allColumnWidths`, `tableLocation`, `tableSize` which has now become local variables.